lib/Docs/SearchIndexer.php

```diff
@@ -63,23 +62,20 @@ public function buildSearchIndexes(
         $records = [];
 
         foreach ($documents as $document) {
-            $this->buildDocumentSearchRecords($document, $records, $project, $version);
+            foreach ($this->buildDocumentSearchRecords($document, $project, $version) as $record) {
+                $records[] = $record;
+            }
         }
 
         $this->getSearchIndex()->saveObjects($records, ['autoGenerateObjectIDIfNotExist' => true]);
     }
 
-    /** @param mixed[][] $records */
```

Comment:
I'd still document what is generated.

`/** Generator<mixed[]> */`

Reply:
I added phpstan types to document the array shapes

Reply:
I added phpstan types to document the array shapes